### DISCUSSION TITLE
Add an option to permanently mute a specific notification

### DISCUSSION THREAD

**Discussion by jxwaa** [ppy/osu#24915] [Ideas] [answered]
Every time I start osu!lazer I get a notification about a tablet error. This notification does nothing to help me, and my tablet works fine. I would be very grateful if somebody added a button below the notification that allowed me to mute it.

**Comment by peppy** [comment]
We'd be looking to fix the error rather than mute it. Could you please provide logs?

**Reply by jxwaa** [reply]
I assume it has something to do with the "usb device not recoginzed" error I sometimes get on Windows when I plug my tablet in. The error happens every time I start lazer. 
[input.log](https://github.com/ppy/osu/files/12711081/input.log)

**Reply by peppy** [reply]
Could you please also post `runtime.log`

**Reply by jxwaa** [reply]
[runtime.log](https://github.com/ppy/osu/files/12721775/runtime.log)
super sorry for the late response

**Reply by peppy** [reply]
cc/ @infinityghost / @x9void for this one, seems like this maybe shouldn't be at error/important level logging?

```
2023-09-25 04:30:23 [verbose]: [Tablet] Detect: Searching for tablets...
2023-09-25 04:30:23 [important]: [Tablet] Device: Failed to set device feature: 02-02
2023-09-25 04:30:23 [verbose]: [Tablet] Detect: Found tablet 'Wacom CTL-472'
2023-09-25 04:30:23 [verbose]: [Tablet] Tablet detected (vid1386), searching for usable configuration...
2023-09-25 04:30:23 [verbose]: [Tablet] Detect: Searching for tablets...
2023-09-25 04:30:23 [important]: [Tablet] Device: Failed to set device feature: 02-02
2023-09-25 04:30:23 [verbose]: [Tablet] Detect: Found tablet 'Wacom CTL-472'
```

**Reply by X9VoiD** [reply]
It's an error that means that the user still has wacom drivers installed. This setup is something that OTD might remove support for in the future since it only gives headache when wacom driver is installed but not running. (Sometimes it also happens that wacom software is uninstalled but remnants remain which also causes this)

When wacom driver is installed, it replaces the system driver for a specific endpoint of the device with its own. This is what causes the "Failed to set device feature" log. Wacom blocks this specific packet which is meant to initialize the device to a mode that OTD (and the wacom driver) supports. If the user mistakenly disables/stops wacom driver thinking it's enough to get it working only through OTD, the blocking of the "feature" packet means device is stuck on an unsupported mode on next reboot/replug.

Will reduce the importance of the log but should be noted that the "correct" solution there is to uninstall wacom driver and do a pass of [TabletDriverCleanup](https://github.com/X9VoiD/TabletDriverCleanup) to ensure no remnant is left.

**Reply by peppy** [reply]
Hmm, I see.

is it possible to improve the log message here so that the user is aware of this? I'd expect it to read something like "While your tablet is supported, we have detected Wacom drivers running in the background. Please uninstall these for full support" (maybe with a URL with further information).

**Reply by X9VoiD** [reply]
There's this [code block](https://github.com/OpenTabletDriver/OpenTabletDriver/blob/543e76b08026ed405ee105eff93e97366687a7e6/OpenTabletDriver.Daemon/DriverDaemon.cs#L107-L114) which can probably be moved to the core library (and bring down log level to warning).

```cs
            foreach (var driverInfo in DriverInfo.GetDriverInfos())
            {
                Log.Write("Detect", $"Another tablet driver found: {driverInfo.Name}", LogLevel.Warning);
                if (driverInfo.IsBlockingDriver)
                    Log.Write("Detect", $"Detection for {driverInfo.Name} tablets might be impaired", LogLevel.Warning);
                else if (driverInfo.IsSendingInput)
                    Log.Write("Detect", $"Detected input coming from {driverInfo.Name} driver", LogLevel.Error);
            }
```

**Reply by peppy** [reply, marked as answer]
Sounds good. I've also made a [tentative PR](https://github.com/ppy/osu-framework/pull/6006) to stop showing `Important` log level to users via notifications, thoughts on this?

**Reply by X9VoiD** [reply]
Should be good, OTD's logs aren't exactly user-friendly right now. So until there's improvement on our side, doing that is beneficial in reducing noise imo.
